Source/WebKit/UIProcess/WebAuthentication/fido/CtapAuthenticator.cpp

```diff
@@ -129,38 +129,51 @@ void CtapAuthenticator::makeCredential()
             ASSERT(RunLoop::isMain());
             if (!weakThis)
                 return;
-            weakThis->continueCheckExcludedCredentialsAfterResponseRecieved(WTFMove(data));
+            weakThis->continueSlientlyCheckCredentials(WTFMove(data), [weakThis = WTFMove(weakThis)] (bool foundMatch) mutable {
+                if (!weakThis)
+                    return;
+                weakThis->continueMakeCredentialAfterCheckExcludedCredentials(foundMatch);
+            });
         });
     } else
         continueMakeCredentialAfterCheckExcludedCredentials();
 }
 
-void CtapAuthenticator::continueCheckExcludedCredentialsAfterResponseRecieved(Vector<uint8_t>&& data)
+void CtapAuthenticator::continueSlientlyCheckCredentials(Vector<uint8_t>&& data, CompletionHandler<void(bool)>&& completionHandler)
 {
     auto error = getResponseCode(data);
     if (error == CtapDeviceResponseCode::kSuccess)
-        return continueMakeCredentialAfterCheckExcludedCredentials(true);
+        return completionHandler(true);
     if (error == CtapDeviceResponseCode::kCtap2ErrNoCredentials) {
         m_currentBatch += 1;
         if (m_currentBatch >= m_batches.size())
-            return continueMakeCredentialAfterCheckExcludedCredentials();
+            return completionHandler(false);
     }
-    auto& options = std::get<PublicKeyCredentialCreationOptions>(requestData().options);
+    Vector<uint8_t> cborCmd;
+    WTF::switchOn(requestData().options, [&](const PublicKeyCredentialCreationOptions& options) {
+        cborCmd = encodeSilentGetAssertion(*options.rp.id, requestData().hash, m_batches[m_currentBatch], std::nullopt);
+    }, [&](const PublicKeyCredentialRequestOptions& options) {
+        cborCmd = encodeSilentGetAssertion(options.rpId, requestData().hash, m_batches[m_currentBatch], std::nullopt);
+    });
 
     auto response = readCTAPGetAssertionResponse(data, AuthenticatorAttachment::CrossPlatform);
-    Vector<uint8_t> cborCmd = encodeSilentGetAssertion(*options.rp.id, requestData().hash, m_batches[m_currentBatch], std::nullopt);
-    driver().transact(WTFMove(cborCmd), [weakThis = WeakPtr { *this }](Vector<uint8_t>&& data) {
+    driver().transact(WTFMove(cborCmd), [weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler)](Vector<uint8_t>&& data) mutable {
         ASSERT(RunLoop::isMain());
         if (!weakThis)
-            return;
-        weakThis->continueCheckExcludedCredentialsAfterResponseRecieved(WTFMove(data));
+            return completionHandler(false);
+        weakThis->continueSlientlyCheckCredentials(WTFMove(data), WTFMove(completionHandler));
     });
 }
 
 void CtapAuthenticator::continueMakeCredentialAfterCheckExcludedCredentials(bool includeCurrentBatch)
 {
     Vector<uint8_t> cborCmd;
-    auto& options = std::get<PublicKeyCredentialCreationOptions>(requestData().options);
+    auto options = std::get<PublicKeyCredentialCreationOptions>(requestData().options);
```

Comment:
This is a fairly large options object to change from a reference to a copy. Are we trying to avoid changing the value stored in `requestData().options`?